test/material/testenv/docker-compose.yaml

```diff
@@ -27,51 +27,86 @@ services:
-#  # ------------------------------
-#  dask-scheduler:
-#    build: # Build the latest SuperDuperDB code
-#      context: ../../../. # Build from project's root
-#      dockerfile: ./images/superduperdb/Dockerfile
-#    hostname: scheduler
-#    ports:
-#      - "8786:8786" # Peer communication
-#      - "8787:8787" # HTTP Dashboard
-#    volumes:
-#        - ../../../:/app # mount the root path
-#    command:
-#      - /bin/sh
-#      - -c
-#      - |
-#        cd /app
-#
-#        # Basic stuff needed for pickling.
-#        # No complex dependencies
-#        pip install .
-#        dask scheduler
-#
-#
-#  dask-worker:
-#    depends_on:
-#      -   dask-scheduler
+
+  # Vector Search Service
+  # ------------------------------
+#  vector-search:
 #    build: # Build the latest SuperDuperDB code
 #      context: ../../../. # Build from project's root
 #      dockerfile: ./images/superduperdb/Dockerfile
+#    hostname: vector-search
 #    volumes:
 #        - ../../../:/app # mount the root path
-##    environment:
-##        - SUPERDUPERDB_DATA_BACKEND = 'mongodb://mongodb:27017'
+#    ports:
+#      - "8000:8000"
 #    command:
 #      - /bin/sh
 #      - -c
 #      - |
 #        cd /app
-#
+
 #        # Require all computational dependencies
-#        pip install .[torch,apis,docs,quality,testing]
-#        dask worker "tcp://scheduler:8786"
-#    deploy:
-#      replicas: 1
+#        pip install --user .[server,apis]
+#        python -m superduperdb vector-search
+#    healthcheck:
+#      test: curl http://localhost:8000/ || exit 1
+#      interval: 40s
+#      timeout: 30s
+#      retries: 10
+#      start_period: 60s
+
+#  vector-search-init: # Once finished, Vector Search is assumed to be ready.
+#    image: busybox
+#    depends_on:
+#      vector-search:
+#        condition: service_healthy
+
+  # Dask Parallel Computation
+  # ------------------------------
+  dask-scheduler:
+    build: # Build the latest SuperDuperDB code
+      context: ../../../. # Build from project's root
+      dockerfile: ./images/superduperdb/Dockerfile
+      args:
+        - BUILD_ENV=sandbox # Use local code
+        - SUPERDUPERDB_EXTRAS=dev # Install dependencies
+    hostname: scheduler
+    ports:
+      - "8786:8786" # Peer communication
+      - "8787:8787" # HTTP Dashboard
+    volumes:
+        - ../../../:/app # mount the root path
+    command:
+      - /bin/sh
+      - -c
+      - |
+        # Install the latest version of the code (avoid cached layers).
+        cd /app; pip install --user .        
+        dask scheduler
+
+
+  dask-worker:
+    depends_on:
+      -   dask-scheduler
+    build: # Build the latest SuperDuperDB code
+      context: ../../../. # Build from project's root
+      dockerfile: ./images/superduperdb/Dockerfile
+      args:
+        - BUILD_ENV=sandbox # Load the local code
+        - SUPERDUPERDB_EXTRAS=dev # Install dependencies
+    volumes:
+        - ../../../:/app # mount the root path
+#    environment:
```

Comment:
can we remove commented code? it looks very experimental code

Reply:
Fixed